### PR TITLE
Some build cleanup preparation

### PR DESCRIPTION
**Pull request by martinpitt**
Broken out from #18519. Lis and I want to do some more changes there, and I don't want that to become too big. ~Also, @KKoukiou is waiting for that PO plugin change for esbuild~ (taken out, causes regression)

### CODE REVIEW

**Comment by martinpitt** [comment]
Meh, the PO plugin update [breaks TestPages](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18522-20230317-170305-147b71f8-rhel-8-8/log.html#124-2). It was happy in #18519, so I'll take it out of this PR for now.

File: tools/webpack-jumpstart [hunk 1]
```diff
@@ -172,11 +172,19 @@ fi
...
-    # remove any webpacks that mention this file.  `grep -s` in case there aren't any left.
-    grep -sl "${file}" dist/*/Makefile.deps | cut -f2 -d/ | while read -r webpack; do
-        message REMOVE "dist/${webpack}"
-        rm -rf "dist/${webpack}"
-    done
-done
+# for --partial, remove webpacks which changed since the jumpstart was built
+if [ -n "${partial}" ]; then
+    # assume that pkg/lib and build system changes affect all of them
+    if echo "$changed_files" | grep -q -E '(pkg/lib|Makefile|package.*.json|webpack.config.js)'; then
```
Comment (KKoukiou) [inline tools/webpack-jumpstart:178]: noone would ever change the 'Makefile' directory, it's always the `Makefile.am`
Is this missing a '*'?

Comment (martinpitt) [reply tools/webpack-jumpstart:178]: It's not a full match, just substrings. I can make it `Makefile.am` explicitly, but shouldn't make much difference.

**Review by KKoukiou — APPROVED** [review]

File: src/appstream/org.cockpit-project.cockpit-kdump.metainfo.xml.in
Comment (allisonkarlitskaya) [inline src/appstream/org.cockpit-project.cockpit-kdump.metainfo.xml.in:1]: This is a better place, thanks.

It also means that the webpack isn't repeated if we ever changed the appstream files, which is also conceptually cleaner.

File: tools/webpack-jumpstart (hunk 1 above)
Comment (allisonkarlitskaya) [reply tools/webpack-jumpstart:178]: How would you feel about sacrificing partial mode, instead?

**Comment by martinpitt** [comment]
> How would you feel about sacrificing partial mode, instead?

It's rather nice for speeding up CI and local development, so I thought I'd rather keep it around until we actually switch to esbuild. But if that heuristic makes you too uneasy, I'd rather drop partial mode and land my stack of PRs, as that is currently blocking esbuild.

**Pull request closed without merging**